language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/foldingrange/FoldingRangeProvider.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.foldingrange;
+
+import io.ballerina.compiler.syntax.tree.ImportDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import org.eclipse.lsp4j.FoldingRange;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Provider class of folding ranges.
+ */
+public class FoldingRangeProvider {
+
+    /**
+     * Returns the list of folding ranges for the given syntax tree.
+     *
+     * @param syntaxTree Syntax tree representation
+     * @return The list of folding ranges
+     */
+    public static List<FoldingRange> getFoldingRange(SyntaxTree syntaxTree) {
+
+        ModulePartNode modulePartNode = syntaxTree.rootNode();
+        FoldingRangeFinder foldingRangeFinder = new FoldingRangeFinder();
+        List<FoldingRange> foldingRangeList = new ArrayList<>(foldingRangeFinder.getFoldingRange(modulePartNode));
+
+        List<ImportDeclarationNode> imports = modulePartNode.imports().stream().collect(Collectors.toList());
```

Comment:
shall we try to handle the imports within the `FoldingRangeFinder`? Otherwise we have to expose the `createFoldingRange` utility method as well as a part of the syntax tree visiting is disconnected

Reply:
Updated.